src/main/java/io/jenkins/plugins/artifact_manager_s3/JCloudsArtifactManager.java

```diff
@@ -76,6 +77,7 @@
 /**
  * Artifact manager that stores files in a JClouds BlobStore using any of JClouds supported backends
  */
+@Restricted(Beta.class)
```

Comment:
IIUC this is not supposed to be an API of any kind.

I would just restrict everything to internal use. We have not really defined how other clouds would be integrated. S3 is hard-coded for now.

Reply:
Well, the “safe” would be

```java
@Restricted(NoExternalUse.class)
```

Since we have not defined & tested any mechanism for a separate plugin to contribute a new artifact manager based on another cloud provider—from my reading of the current implementation, it is impossible until we do some refactoring—we should just block it.

I would go so far as to make `JCloudsApiExtensionPoint` package-private and not use `Beta` at all. We can define externally consumable APIs, marked as beta, if and when we have at least one candidate implementation using something other than S3.

Reply:
Well, I want to let people play with it, just no warranties, so that's what @Beta should be for